scripts/build_test.py

```diff
@@ -14,29 +14,313 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import StringIO
+import collections
+import os
+import random
+import re
+import subprocess
+import threading
+
 # pylint: disable=relative-import
 import build
 from core.tests import test_utils
 
 # pylint: enable=relative-import
 
+TEST_DIR = os.path.join('core', 'tests', 'build', '')
+TEST_SOURCE_DIR = os.path.join('core', 'tests', 'build_sources')
+
+ASSETS_DEV_DIR = os.path.join(TEST_SOURCE_DIR, 'assets', '')
```

Comment:
Let's name the stuff in TEST_SOURCE_DIR: MOCK_ASSETS_DEV_DIR, MOCK_EXTENSIONS_DEV_DIR, MOCK_TEMPLATES_DEV_DIR.

That's a good reminder to convey to the reader that we're not using the actual dirs, just mocks.

Reply:
Yup 100% in agreement here. I was just filling the blank when I made all these global variables. Global var renamed.